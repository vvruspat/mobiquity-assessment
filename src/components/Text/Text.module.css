.size_s {
    font-size: var(--font-size-s);
}

.size_m {
    font-size: var(--font-size-m);
}

.size_l {
    font-size: var(--font-size-l);
}

.size_xl {
    font-size: var(--font-size-xl);
}

.size_xxl {
    font-size: var(--font-size-xxl);
}

.regular {
    font-weight: var(--font-weight-regular);
}

.semibold {
    font-weight: var(--font-weight-semibold);
}

.bold {
    font-weight: var(--font-weight-bold);
}