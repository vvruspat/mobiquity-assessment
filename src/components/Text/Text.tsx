import { FC, PropsWithChildren } from "react";
import cn from "clsx";

import s from "./Text.module.css";

type TextProps = {
  /** Text size */
  size?: "s" | "m" | "l" | "xl" | "xxl";
  /** Text weight */
  weight?: "regular" | "bold" | "semibold";
};

/**
 * Text component
 */
export const Text: FC<PropsWithChildren<TextProps>> = ({
  children,
  size = "m",
  weight = "regular",
}) => {
  return (
    <span className={cn(s.text, [s[`size_${size}`]], [s[weight]])}>
      {children}
    </span>
  );
};
