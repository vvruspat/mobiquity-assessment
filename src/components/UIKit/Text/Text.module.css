.size_s {
    font-size: var(--font-size-s);
    line-height: var(--line-height-s);
}

.size_m {
    font-size: var(--font-size-m);
    line-height: var(--line-height-m);
}

.size_l {
    font-size: var(--font-size-l);
    line-height: var(--line-height-l);
}

.size_xl {
    font-size: var(--font-size-xl);
    line-height: var(--line-height-xl);
}

.size_xxl {
    font-size: var(--font-size-xxl);
    line-height: var(--line-height-xxl);
}

.regular {
    font-weight: var(--font-weight-regular);
}

.semibold {
    font-weight: var(--font-weight-semibold);
}

.bold {
    font-weight: var(--font-weight-bold);
}