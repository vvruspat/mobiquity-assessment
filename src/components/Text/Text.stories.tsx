import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Text } from "./Text";
import "../../index.css";

export default {
  title: "UI Kit/Typography/Text",
  component: Text,
} as ComponentMeta<typeof Text>;

const Template: ComponentStory<typeof Text> = (args) => (
  <Text {...args}>Text</Text>
);

export const TextTemplate = Template.bind({});
TextTemplate.args = {};
