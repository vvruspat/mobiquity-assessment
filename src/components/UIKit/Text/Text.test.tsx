import { render, screen } from "@testing-library/react";
import { Text } from "./Text";

test("renders text", () => {
  render(<Text>Test text</Text>);

  const el = screen.getByTestId("text");

  expect(el).toHaveClass("text");
  expect(el).toHaveClass("size_m");
  expect(el).toHaveClass("regular");
  expect(el).toHaveTextContent("Test text");
  expect(el).toMatchSnapshot();
});

test("renders text size s", () => {
  render(<Text size="s">Test text</Text>);

  const el = screen.getByTestId("text");

  expect(el).toHaveClass("text");
  expect(el).toHaveClass("size_s");
  expect(el).toHaveClass("regular");
  expect(el).toHaveTextContent("Test text");
  expect(el).toMatchSnapshot();
});

test("renders text size l", () => {
  render(<Text size="l">Test text</Text>);

  const el = screen.getByTestId("text");

  expect(el).toHaveClass("text");
  expect(el).toHaveClass("size_l");
  expect(el).toHaveClass("regular");
  expect(el).toHaveTextContent("Test text");
  expect(el).toMatchSnapshot();
});

test("renders text size xl", () => {
  render(<Text size="xl">Test text</Text>);

  const el = screen.getByTestId("text");

  expect(el).toHaveClass("text");
  expect(el).toHaveClass("size_xl");
  expect(el).toHaveClass("regular");
  expect(el).toHaveTextContent("Test text");
  expect(el).toMatchSnapshot();
});

test("renders text size xxl", () => {
  render(<Text size="xxl">Test text</Text>);

  const el = screen.getByTestId("text");

  expect(el).toHaveClass("text");
  expect(el).toHaveClass("size_xxl");
  expect(el).toHaveClass("regular");
  expect(el).toHaveTextContent("Test text");
  expect(el).toMatchSnapshot();
});

test("renders text bold", () => {
  render(<Text weight="bold">Test text</Text>);

  const el = screen.getByTestId("text");

  expect(el).toHaveClass("text");
  expect(el).toHaveClass("size_m");
  expect(el).toHaveClass("bold");
  expect(el).toHaveTextContent("Test text");
  expect(el).toMatchSnapshot();
});

test("renders text semibold", () => {
  render(<Text weight="semibold">Test text</Text>);

  const el = screen.getByTestId("text");

  expect(el).toHaveClass("text");
  expect(el).toHaveClass("size_m");
  expect(el).toHaveClass("semibold");
  expect(el).toHaveTextContent("Test text");
  expect(el).toMatchSnapshot();
});
