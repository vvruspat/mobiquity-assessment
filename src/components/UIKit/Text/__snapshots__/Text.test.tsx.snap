// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text 1`] = `
<span
  class="text size_m regular"
  data-testid="text"
>
  Test text
</span>
`;

exports[`renders text bold 1`] = `
<span
  class="text size_m bold"
  data-testid="text"
>
  Test text
</span>
`;

exports[`renders text semibold 1`] = `
<span
  class="text size_m semibold"
  data-testid="text"
>
  Test text
</span>
`;

exports[`renders text size l 1`] = `
<span
  class="text size_l regular"
  data-testid="text"
>
  Test text
</span>
`;

exports[`renders text size s 1`] = `
<span
  class="text size_s regular"
  data-testid="text"
>
  Test text
</span>
`;

exports[`renders text size xl 1`] = `
<span
  class="text size_xl regular"
  data-testid="text"
>
  Test text
</span>
`;

exports[`renders text size xxl 1`] = `
<span
  class="text size_xxl regular"
  data-testid="text"
>
  Test text
</span>
`;
